param/param.go

```diff
@@ -121,3 +123,70 @@ func ConvertToConfigWithType(rawConfig *Config) *configWithType {
 	convertStruct(srcVal, destVal)
 	return &typedConfig
 }
+
+// getConfigValue retrieves a value from the Config struct using a dotted path.
+// For example, "Cache.DataLocation" returns config.Cache.DataLocation.
+// This is a helper function used by the generated accessor methods.
+func getConfigValue(config *Config, path string) (interface{}, bool) {
```

Comment:
I don't like this new approach.  `getConfigValue` looks computationally complex.

Because `generate/param_generator.go` is writing code, there should be no need for reflection.  Instead, pre-compute the location within the config struct based on the actual configuration name and access it directly within the generated code.

This should all be doable without a helper.